internal/services/controller/controller_test.go

```diff
@@ -242,6 +243,157 @@ func TestController_ApiResourcesErrorProcessing(t *testing.T) {
 	}])
 }
 
+func TestController_ShouldSendByInterval(t *testing.T) {
+	tt := []struct {
+		name          string
+		sendInterval  time.Duration
+		sendDurations []time.Duration
+		checkAfter    time.Duration
+		wantSends     int64
+	}{
+		{
+			name:         "should trigger all sends when none exceed allocated intervals",
+			sendInterval: 300 * time.Millisecond,
+			sendDurations: []time.Duration{
+				// Send by 300ms
+				250 * time.Millisecond,
+				// ...by 600ms
+				250 * time.Millisecond,
+				// ...by 900ms
+				250 * time.Millisecond,
+				// ...by 1200ms
+				250 * time.Millisecond,
+			},
+			checkAfter: 1200 * time.Millisecond,
+			wantSends:  4,
+		},
+		{
+			name:         "should trigger all sends when previous send exceeds one interval",
+			sendInterval: 300 * time.Millisecond,
+			sendDurations: []time.Duration{
+				// At 0, 300: idle, previous still sending
+				// Send by 600 ms
+				450 * time.Millisecond,
+				// Send by 600 ms too, since we expect ticker to fire events even between ticks
+				50 * time.Millisecond,
+			},
+			checkAfter: 600 * time.Millisecond,
+			wantSends:  2,
+		},
+		{
+			name:         "should trigger all sends when previous send exceeds two intervals",
```

Comment:
@saumas 

Here is a test case where a request is skipped because the first one takes longer to send a request than the non-sliding interval.

Reply:
There's also a test case you might be interested in at line 277. Currently, we launch a queued send request immediately after unblocking without dropping it because the `wait` package uses a `time.Ticker` under the hood.